src/zenml/cli/stack.py

```diff
@@ -176,87 +154,86 @@ def stack() -> None:
-            StackComponentType.CONTAINER_REGISTRY: container_registry_name,
-            StackComponentType.DATA_VALIDATOR: data_validator_name,
-            StackComponentType.EXPERIMENT_TRACKER: experiment_tracker_name,
-            StackComponentType.FEATURE_STORE: feature_store_name,
-            StackComponentType.MODEL_DEPLOYER: model_deployer_name,
-            StackComponentType.ORCHESTRATOR: orchestrator_name,
-            StackComponentType.SECRETS_MANAGER: secrets_manager_name,
-            StackComponentType.STEP_OPERATOR: step_operator_name,
-        }
-        stack_components = dict()
-
-        for c_type, name in type_component_mapping.items():
-            if name:
-                stack_components[c_type] = [
-                    cli_utils.get_component_by_id_or_name_or_prefix(
-                        client=client,
-                        component_type=c_type,
-                        id_or_name_or_prefix=name,
-                    ).id
-                ]
+        components = dict()
+
+        components[StackComponentType.ARTIFACT_STORE] = artifact_store
+        components[StackComponentType.ORCHESTRATOR] = orchestrator
+
+        if alerter:
+            components[StackComponentType.ALERTER] = alerter
+        if annotator:
+            components[StackComponentType.ANNOTATOR] = annotator
+        if data_validator:
+            components[StackComponentType.DATA_VALIDATOR] = data_validator
+        if feature_store:
+            components[StackComponentType.FEATURE_STORE] = feature_store
+        if model_deployer:
+            components[StackComponentType.MODEL_DEPLOYER] = model_deployer
+        if secrets_manager:
+            components[StackComponentType.SECRETS_MANAGER] = secrets_manager
+        if step_operator:
+            components[StackComponentType.STEP_OPERATOR] = step_operator
+        if experiment_tracker:
+            components[
+                StackComponentType.EXPERIMENT_TRACKER
+            ] = experiment_tracker
+        if container_registry:
+            components[
+                StackComponentType.CONTAINER_REGISTRY
+            ] = container_registry
 
         # click<8.0.0 gives flags a default of None
         if share is None:
             share = False
 
-        from zenml.models import StackModel
-
-        stack_ = StackModel(
+        created_stack = client.register_stack(
             name=stack_name,
-            components=stack_components,
+            components=components,
             is_shared=share,
-            project=client.active_project.id,
-            user=client.active_user.id,
         )
 
-        created_stack = client.register_stack(stack_)
-        cli_utils.declare(f"Stack '{stack_name}' successfully registered!")
+        cli_utils.declare(
+            f"Stack '{created_stack.name}' successfully registered!"
+        )
 
     if set_stack:
-        _set_active_stack(created_stack)
+        client.activate_stack(created_stack.id)
```

Comment:
The point of this helper function was to log something to give some feedback that the stack was activated. This is missing now

Reply:
You are right. I have had the logs already on the `stack set` but not on the `stack register --set`. I have fixed it and pushed it.